File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/api/dto/Edition.java
package ru.elenacherev.librarymanager.api.dto;

import lombok.Builder;
import lombok.Getter;
import ru.elenacherev.librarymanager.util.AgeRating;

import java.util.List;

@Builder
@Getter
public class Edition {
    private Long editionId;
    
    private String title;
    private String workTitle; //
    private int year;
    private int publishingYear;
    private String isbn;
    private boolean isIllustrated;
    private String downloadLink;
    private AgeRating age;
    private String info;
    
    private String langString;
    private Long langId;
    
    private String publLangString;
    private Long publLangId;
    
    private String publishingHouseString;
    private Long publishingHouseId;
    
    private String genreString;
    private Long genreId;
    
    private String sectionTitle;
    private Long sectionId;
    
    private List<Long> authorIds;
    private List<String> authorFIOs;
}

File: librarymanager-api/src/main/java/ru/elenacherev/librarymanager/api/dto/Edition.kt
package ru.elenacherev.librarymanager.api.dto

import ru.elenacherev.librarymanager.api.enum.AgeRating
import java.util.*

data class Edition(
    var editionId: UUID? = null,
    val title: String,
    val workTitle: String,
    val year: Int,
    val publishingYear: Int = 0,
    val isbn: String,
    val isIllustrated: Boolean = false,
    val downloadLink: String? = null,
    val age: AgeRating,
    val info: String? = null,
    val langString: String,
    val langId: UUID,
    val publLangString: String,
    val publLangId: UUID,
    val publishingHouseString: String,
    val publishingHouseId: UUID,
    val genreString: String,
    val genreId: UUID,
    val sectionTitle: String,
    val sectionId: UUID,
    val authorIds: List<UUID>,
    val authorFIOs: List<String>
)